internal/cache/controller_obj_cache.go

```diff
@@ -3461,6 +3461,40 @@ func validateAndConfigureSeGroup(client *clients.AviClient, returnErr *error) bo
 	return true
 }
 
+func RefreshSeGroupDataAndCheckLabel(client *clients.AviClient, seGroup *models.ServiceEngineGroup) error {
+	segName := *seGroup.Name
+	uri := "/api/serviceenginegroup/?include_name&name=" + segName
+	result, err := lib.AviGetCollectionRaw(client, uri)
+	if err != nil {
+		utils.AviLog.Warnf("ServiceEngineGroup Get uri %v returned err %v", uri, err)
+		return err
+	}
+
+	elems := make([]json.RawMessage, result.Count)
+	err = json.Unmarshal(result.Results, &elems)
+	if err != nil {
+		utils.AviLog.Warnf("Failed to unmarshal data, err: %v", err)
+		return err
+	}
+
+	if result.Count != 1 {
+		utils.AviLog.Errorf("ServiceEngineGroup not found for ServiceEngineGroup name: %s", segName)
+		return err
+	}
+
+	serviceEngineGroup := models.ServiceEngineGroup{}
+	if err = json.Unmarshal(elems[0], &serviceEngineGroup); err != nil {
+		utils.AviLog.Warnf("Failed to unmarshal serviceenginegroup data, err: %v", err)
+		return err
+	}
+	segLabelEq := reflect.DeepEqual(seGroup.Labels, lib.GetLabels())
```

Comment:
A membership check would have been better in IMO as an SE group may have other labels also, but it's not a must fix